src/datepicker/datepicker.component.ts

```diff
@@ -213,13 +218,27 @@ export class DatePicker implements OnDestroy, OnChanges, AfterViewChecked {
 		}
 	}
 
+	ngAfterViewInit() {
+		setTimeout(() => {
+			if (this.isFlatpickrLoaded() && this.elementService) {
```

Comment:
`this.elementService` should be available right away ... there's no subclasses of datepicker and it's not optional so we should be fine to just check if it's loaded

Reply:
The loaded check should also likely go inside the subscribe ... we can't know that `isFlatpickrLoaded` will be true by the time the timeout and `AfterViewInit` run